care/facility/api/serializers/patient_consultation.py

```diff
@@ -604,16 +604,28 @@ def validate(self, attrs):  # noqa: PLR0912
         validated = super().validate(attrs)
         # TODO Add Bed Authorisation Validation
 
+        facility = (
+            self.instance and self.instance.facility or validated["patient"].facility
+        )
+
         if (
             not self.instance or validated.get("patient_no") != self.instance.patient_no
         ) and "suggestion" in validated:
             suggestion = validated["suggestion"]
             patient_no = validated.get("patient_no")
 
-            if suggestion == SuggestionChoices.A and not patient_no:
-                raise ValidationError(
-                    {"patient_no": "This field is required for admission."}
-                )
+            if suggestion == SuggestionChoices.A:
+                if not patient_no:
+                    raise ValidationError(
+                        {"patient_no": "This field is required for admission."}
+                    )
+                result_count = PatientConsultation.objects.filter(
+                    facility=facility, patient_no=patient_no
+                ).count()
```

Comment:
we've had issue with readmissions using this validation
https://github.com/ohcnetwork/care_fe/issues/7798